lib/AST/ASTDumper.cpp

```diff
@@ -1270,11 +1323,25 @@ namespace {
 
       printField(requirement.getKind(), Label::optional("kind"));
 
-      if (requirement.getKind() != RequirementKind::Layout
-            && requirement.getSecondType())
-        printTypeField(requirement.getSecondType(),
-                       Label::optional("second_type"), opts);
-      else if (requirement.getLayoutConstraint())
+      if (requirement.getKind() != RequirementKind::Layout &&
+          requirement.getSecondType()) {
```

Comment:
FWIW, getSecondType() always returns a non-null value so you don't need to check it